### PR TITLE
Fix got wrong numa node in netperf.py

### PR DESCRIPTION
**Pull request by mazhang**
Signed-off-by: Maosheng Zhang <mazhang@redhat.com>

### CODE REVIEW

**Comment by mazhang** [comment]
ID: 1315167

**Comment by MiriamDeng** [comment]
Acked by Miriam Deng.Run the patch to get the correct log info,thanks.
03/17 05:51:05 DEBUG|   netperf:0158| available: 4 nodes (0-1,16-17)
node 0 cpus: 0 8 16 24 32
node 0 size: 65536 MB
node 0 free: 8452 MB
node 1 cpus: 40 48 56 64 72
node 1 size: 65536 MB
node 1 free: 4042 MB
node 16 cpus: 80 88 96 104
node 16 size: 65536 MB
node 16 free: 6026 MB
node 17 cpus: 120 128 136 144 152
node 17 size: 65536 MB
node 17 free: 22874 MB
node distances:
...
03/17 05:51:41 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (0.000008 secs)
03/17 05:51:41 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:41 INFO |   netperf:0539| Start netperf thread by cmd 'numactl --cpunodebind=17 --membind=17 /tmp/netperf_agent.py 1 /tmp/netperf-2.6.0/src/netperf -D 1 -H 10.16.70.228 -l 90.0 -C -c -t TCP_STREAM -- -m 64 >> /tmp/netperf.123549.nf'
03/17 05:51:41 DEBUG|     utils:0114| Running 'numactl --cpunodebind=17 --membind=17 /tmp/netperf_agent.py 1 /tmp/netperf-2.6.0/src/netperf -D 1 -H 10.16.70.228 -l 90.0 -C -c -t TCP_STREAM -- -m 64 >> /tmp/netperf.123549.nf'
03/17 05:51:41 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (0.205546 secs)
03/17 05:51:41 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:41 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (0.411627 secs)
03/17 05:51:41 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (0.617089 secs)
03/17 05:51:42 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (0.822305 secs)
03/17 05:51:42 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (1.027472 secs)
03/17 05:51:42 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (1.232984 secs)
03/17 05:51:42 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|qemu_monit:0282| (monitor catch_monitor) Sending command 'info registers' (via Human Monitor)
03/17 05:51:42 DEBUG|utils_misc:0510| Wait until all netperf clients start to work (1.438893 secs)
03/17 05:51:42 DEBUG|     utils:0114| Running 'cat /tmp/netperf.123549.nf'
03/17 05:51:42 DEBUG|   netperf:0599| All netperf clients start to work.
03/17 05:51:42 DEBUG|   aexpect:1465| Sending command: ifconfig
03/17 05:51:42 DEBUG|   aexpect:1465| Sending command: echo $?

**Comment by Zhengtong** [comment]
Acked-by: Zhengtong Liu <zhengtli@redhat.com>

22:54:07 INFO | Start netperf thread by cmd 'numactl --cpunodebind=17 --membind=17 /tmp/netperf_agent.py 1 /tmp/netperf-2.6.0/src/netperf -D 1 -H 10.19.116.212 -l 90.0 -C -c -t TCP_STREAM -- -m 64 >> /tmp/netperf.15639.nf'
22:55:10 INFO | Context: Start netperf testing --> Testing TCP_STREAM protocol --> Testing Results Treatment and Report
22:55:10 INFO |         size|    sessions|  throughput|         CPU| thr_per_CPU|     rx_pkts|     tx_pkts|     rx_byts|     tx_byts|     re_pkts|       exits| rx_intr_sum| tx_intr_sum
          64|           1|      669.95|       13.34|       50.22|     2773910|     1520579|  5225984856|   100338450|           0|     1249485|     1586006|          23
22:55:10 INFO | Netperf thread completed successfully

File: generic/tests/netperf.py [hunk 1]
```diff
@@ -534,8 +534,8 @@ def netperf_thread(i, numa_enable, client_s, timeout):
         cmd = ""
         fname = "/tmp/netperf.%s.nf" % pid
         if numa_enable:
-            output = ssh_cmd(client_s, "numactl --hardware")
-            n = int(re.findall(r"available: (\d+) nodes", output)[0]) - 1
```
Comment (xutian) [inline generic/tests/netperf.py]: use 'numactl --hardware' to get available node looks more suitable.

**Comment by xutian** [comment]
@mazhang  same question with PR#565, can you make a hotfix ASAP please.

**Comment by mazhang** [comment]
Close this PR, merge the fix to #565

**Pull request closed without merging**